mail/common/src/main/java/com/fsck/k9/mail/internet/AddressHeaderBuilder.kt

```diff
@@ -0,0 +1,35 @@
+package com.fsck.k9.mail.internet
+
+import com.fsck.k9.mail.Address
+
+/**
+ * Encode and fold email addresses for use in header values.
+ */
+object AddressHeaderBuilder {
+    @JvmStatic
+    fun createHeaderValue(addresses: Array<Address>): String? {
+        if (addresses.isEmpty()) return null
```

Comment:
:thinking: Maybe it would be better if empty string was returned in this case instead of `null`? `null` requires special handling in all clients.

Reply:
That's intentional. We don't want to add a header with an empty value. If `null` is returned no header must be added.

Reply:
As far as I can see all call sites check the return value (`if (headerValue == null) {`), they could instead check for input to `createHeaderValue` (e.g. `if (addresses.length == 0)`). That would be more obvious that you unset the header when there are no addresses.

Reply:
I updated the code to follow your suggestion.